### PR TITLE
feat(v4-examples): Migrate examples to new auth provider

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
Examples migrated to new auth provider

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: examples/antd-audit-log/src/App.tsx [hunk 1]
```diff
@@ -26,37 +23,59 @@ import refineSDK from "utils/refine-sdk";
...
+const authProvider: AuthBindings = {
     login: async (payload: ILoginDto) => {
         const { email, password, providerName } = payload;
 
-        await refineSDK.auth.login({
-            email,
-            password,
-            provider: providerName,
-        });
+        console.log("worked");
```
Comment (omeraplak) [inline examples/antd-audit-log/src/App.tsx:30]:
```suggestion
```

File: examples/antd-audit-log/src/App.tsx [hunk 2]
```diff
@@ -26,37 +23,75 @@ import refineSDK from "utils/refine-sdk";
...
 
-        return refineSDK.auth
-            .session()
-            .then(() => {
-                return Promise.resolve();
-            })
-            .catch(() => Promise.reject());
+            return {
+                authenticated: true,
+                redirectTo: "/login",
```
Comment (BatuhanW) [inline examples/antd-audit-log/src/App.tsx:77]: Should we redirect here?

File: examples/antd-audit-log/src/App.tsx [hunk 3]
```diff
@@ -26,37 +23,75 @@ import refineSDK from "utils/refine-sdk";
...
+    onError: async () => ({}),
+    check: async () => {
+        try {
+            await refineSDK.auth.getSessionFromUrl();
+            const user = await refineSDK.auth.session();
+
+            if (!user) {
+                return {
+                    authenticated: false,
+                    redirectTo: "/login",
```
Comment (BatuhanW) [inline examples/antd-audit-log/src/App.tsx:71]: Do we need `logout: true` here?

File: examples/data-provider-supabase/src/App.tsx [hunk 4]
```diff
@@ -19,124 +15,221 @@ import "@pankod/refine-antd/dist/reset.css";
...
+        } catch (error: any) {
+            return {
+                success: false,
+                error,
+            };
         }
+
+        return {
+            success: false,
+            error: new Error("Login failed"),
```
Comment (salihozdemir) [inline examples/data-provider-supabase/src/App.tsx:101]:
```suggestion
            error: new Error("Register failed"),
```

File: examples/antd-audit-log/src/App.tsx (hunk 2 above)
Comment (alicanerdurmaz) [reply examples/antd-audit-log/src/App.tsx:77]: yes

File: examples/antd-audit-log/src/App.tsx (hunk 3 above)
Comment (alicanerdurmaz) [reply examples/antd-audit-log/src/App.tsx:71]: I guess yes, depends on the backend tho. I will add `logout: true`

File: examples/with-storybook-mui/src/authProvider.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export const authProvider: AuthProvider = {
 
         return Promise.reject();
     },
-    getPermissions: () => Promise.resolve(),
+    getPermissions: async () => null,
```
Comment (salihozdemir) [inline examples/with-storybook-mui/src/authProvider.ts:28]: The other methods are not updated in this example. Can you update them?

File: examples/with-custom-pages/src/App.tsx [hunk 6]
```diff
@@ -28,25 +29,36 @@ const AuthenticatedPostReview = () => {
...
-                ? Promise.resolve()
-                : Promise.reject(),
-        getPermissions: () => Promise.resolve(["admin"]),
+                ? Promise.resolve({
+                      authenticated: true,
+                  })
+                : Promise.resolve({
+                      authenticated: false,
+                      redirectTo: "/login",
+                  }),
```
Comment (salihozdemir) [inline examples/with-custom-pages/src/App.tsx:60]:
```suggestion
                ? {
                      authenticated: true,
                  }
                : {
                      authenticated: false,
                      redirectTo: "/login",
                  },
```

**Review by salihozdemir — CHANGES_REQUESTED** [review]

File: examples/mern-dashboard-client/src/App.tsx [hunk 7]
```diff
@@ -115,7 +115,7 @@ function App() {
             return Promise.reject();
         },
 
-        getPermissions: () => Promise.resolve(),
+        getPermissions: async () => null,
```
[the remark this answers is not in the document]
Comment (alicanerdurmaz) [reply examples/mern-dashboard-client/src/App.tsx:118]: This example from youtube video. should we change that ?

[the remark this answers is not in the document]
Comment (alicanerdurmaz) [reply examples/mern-dashboard-client/src/App.tsx:118]: This example from youtube video. should we change that ?

[the remark this answers is not in the document]
Comment (salihozdemir) [reply examples/mern-dashboard-client/src/App.tsx:118]: Oh, sorry I missed it. No need, I marked resolve 👌

**Review by salihozdemir — APPROVED** [review]

**Review by aliemir — APPROVED** [review]

**Review by salihozdemir — APPROVED** [review]

**Pull request merged**